File: services/api/app/schemas/control_plane_audit.py
"""Phase 13 control-plane audit API schemas."""

from __future__ import annotations

import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict


class ControlPlaneAuditEventRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    customer_id: uuid.UUID
    site_id: uuid.UUID | None
    actor_user_id: uuid.UUID | None
    action_type: str
    resource_type: str
    resource_id: uuid.UUID | None
    correlation_id: str | None
    payload_json: dict[str, Any] | None
    created_at: datetime


class ControlPlaneAuditEventListResponse(BaseModel):
    items: list[ControlPlaneAuditEventRead]

File: services/frontend/src/api/controlPlaneAudit.ts
import { apiFetch } from "./client";

export type ControlPlaneAuditEventRead = {
  id: string;
  customer_id: string;
  site_id: string | null;
  actor_user_id: string | null;
  action_type: string;
  resource_type: string;
  resource_id: string | null;
  correlation_id: string | null;
  payload_json: Record<string, unknown> | null;
  created_at: string;
};

export async function listControlPlaneAuditEvents(params?: {
  site_id?: string;
  action_type?: string;
  limit?: number;
}) {
  const sp = new URLSearchParams();
  if (params?.site_id) sp.set("site_id", params.site_id);
  if (params?.action_type?.trim()) sp.set("action_type", params.action_type.trim());
  if (params?.limit != null) sp.set("limit", String(params.limit));
  const qs = sp.toString();
  return apiFetch<{ items: ControlPlaneAuditEventRead[] }>(qs ? `/audit/events?${qs}` : "/audit/events");
}
